apps/next/pages/api/stripe/webhook.ts

```diff
@@ -111,11 +117,102 @@ handler.post(async (req, res) => {
+                                        base64Image
+                                    );
+                                    if (profilePicStatus === 200) {
+                                        logger.info('Successfully reset profile picture on subscription deleted.', { userId: userId });
+                                    } else {
+                                        logger.error('Error resetting profile picture on subscription deleted.', { userId: userId });
+                                    }
+                                } else {
+                                    logger.error('could not find a base64 original profile picture image.', { userId: userId });
+                                }
+                                await flipFeatureEnabled(userId, 'profileImage');
+                                logger.info('Profile image disabled on subscription cancelled.', { userId: userId });
+                            }
+
+                            // update the banner to default properties. It is too difficualt to tell what is default and what isn't individually.
+                            // if this returns null, the user never setup a banner. That is fine
+                            const bannerUpdate = await prisma.banner.update({
+                                where: {
+                                    userId: userId,
+                                },
+                                data: {
+                                    userId: userId,
+                                    backgroundId: defaultBannerSettings.backgroundId,
+                                    foregroundId: defaultBannerSettings.foregroundId,
+                                    foregroundProps: defaultBannerSettings.foregroundProps,
+                                    backgroundProps: defaultBannerSettings.backgroundProps,
+                                },
+                            });
+
+                            if (bannerUpdate) {
+                                logger.info(`Banner reset back to default values on subscription cancelled for user: ${userId}`, { userId: userId });
+                            } else {
+                                logger.error(`No banner found on user subscription cancelled. User: ${userId}`, { userId: userId });
+                            }
+
+                            // reset name feature to default value. Get their original name and then append to the front of that
+                            const twitterOriginalName = await prisma.twitterOriginalName.findFirst({
+                                where: {
+                                    userId: userId,
+                                },
+                            });
+
+                            if (twitterOriginalName && twitterOriginalName.originalName) {
+                                const newLiveName = `🔴 Live now | ${twitterOriginalName.originalName}`.substring(0, 50);
```

Comment:
We should probably find a way to share this logic since we use it here and for the actual name feature. I can imagine a scenario where we change it there, then forget to update it here.